### PR TITLE
Remove negative language from Models documentation

### PR DESCRIPTION
**Pull request by amandahla**
## Change Summary

This PR changes the documentation by removing negative language. The revised versions aim to present the information without implying a judgment on developers' choices.

## Checklist

* [X] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [X] Documentation reflects the changes where applicable
* [X] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @hramezani

### CODE REVIEW

File: docs/concepts/models.md [hunk 1]
```diff
@@ -1222,8 +1222,8 @@ values of instance attributes will raise errors. See the [API reference][pydanti
     This config flag is deprecated in Pydantic V2, and has been replaced with `frozen`.
 
 !!! warning
-    Immutability in Python is never strict. If developers are determined/stupid they can always
-    modify a so-called "immutable" object.
+    In Python, strict immutability is not enforced. Developers have the ability to modify objects
```
Comment (samuelcolvin) [inline docs/concepts/models.md:1225]:
```suggestion
    Immutability in Python is never strict. Developers have the ability to modify objects
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise fine by me :shrug:.

File: docs/concepts/models.md (hunk 1 above)
Comment (adriangb) [reply docs/concepts/models.md:1225]: I like the `is not enforced` language better. I find it semantically clearer than `strict`.

**Comment by amandahla** [comment]
please review

File: docs/concepts/models.md (hunk 1 above)
Comment (adriangb) [reply docs/concepts/models.md]: That sounds better to me

File: docs/concepts/models.md [hunk 2]
```diff
@@ -1222,8 +1222,8 @@ values of instance attributes will raise errors. See the [API reference][pydanti
     This config flag is deprecated in Pydantic V2, and has been replaced with `frozen`.
 
 !!! warning
-    Immutability in Python is never strict. If developers are determined/stupid they can always
-    modify a so-called "immutable" object.
+    Immutability in Python is never strict. Developers have the ability to modify objects
```
Comment (sydney-runkle) [inline docs/concepts/models.md:1225]:
```suggestion
    In Python, immutability is not enforced. Developers have the ability to modify objects
```

**Review by sydney-runkle — APPROVED** [review]

**Pull request merged**